### PR TITLE
[#48353] Implement triggers to run background job

### PR DESCRIPTION
**Pull request by ba1ash**
https://community.openproject.org/work_packages/48353

### CODE REVIEW

File: modules/storages/app/models/storages/nextcloud_storage.rb [hunk 1]
```diff
@@ -36,6 +36,17 @@ class Storages::NextcloudStorage < Storages::Storage
 
   alias_method :has_managed_project_folders?, :has_managed_project_folders
 
+  def self.sync_all_group_folders
+    ApplicationRecord.with_advisory_lock('nextcloud_storages_sync', timeout_seconds: 0) do
```
Comment (ba1ash) [inline modules/storages/app/models/storages/nextcloud_storage.rb]: Use `OpenProject::Mutex` here, please.

Comment (ba1ash) [reply modules/storages/app/models/storages/nextcloud_storage.rb]: Add a comment: `Returns false if lock cannot be acquired, block is not executed then.`

File: modules/storages/app/workers/manage_nextcloud_integration_cron_job.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class ManageNextcloudIntegrationCronJob < Cron::CronJob
+  using ::Storages::Peripherals::ServiceResultRefinements
+
+  queue_with_priority :low
+
+  self.cron_expression = '1 * * * *'
```
Comment (ba1ash) [inline modules/storages/app/workers/manage_nextcloud_integration_cron_job.rb]:
```suggestion
  self.cron_expression = '*/5 * * * *'
```

File: modules/storages/spec/workers/manage_nextcloud_integration_cron_job_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require 'spec_helper'
+
+RSpec.describe ManageNextcloudIntegrationCronJob, type: :job do
+  it 'has a schedule set' do
+    expect(described_class.cron_expression).to eq('1 * * * *')
```
Comment (ba1ash) [inline modules/storages/spec/workers/manage_nextcloud_integration_cron_job_spec.rb]: Change it.

File: modules/storages/app/workers/manage_nextcloud_integration_events_job.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class ManageNextcloudIntegrationEventsJob < ApplicationJob
+  def self.debounce
+    debounce_time = 1.minute
```
Comment (ba1ash) [inline modules/storages/app/workers/manage_nextcloud_integration_events_job.rb]: Decrease to 20 seconds.

**Review by apfohl — APPROVED** [review]

**Pull request merged**